feat: support multiple versions of the postgis extension

jfroche opened the pull request:
Build multiple versions of the postgis extension on different PostgreSQL versions.
Add test for the extensions and their upgrade on PostgreSQL 15 and 17.

jfroche committed:
feat: support multiple versions of the postgis extension

Build multiple versions of the postgis extension on different PostgreSQL versions.
Add test for the extensions and their upgrade on PostgreSQL 15 and 17.

hunleyd reviewed (changes requested):
rebase to fix conflicts?

samrose committed:
chore: correct formatting

samrose committed:
fix: symbolic linking removal

samrose committed:
chore: bump version to retest

samrose commented:
Please don’t merge this one. Upgrade tests are failing

samrose reviewed (changes requested):
Upgrade tests failing

samrose committed:
fix:
This fix ensures that the control files reference $libdir/$ext-$MIN_MAJ_VERSION (e.g., $libdir/postgis-3.3) which matches the actual library names built with the
  --with-library-minor-version flag. This allows:

  1. Multiple PostGIS versions to coexist (3.3.2 and 3.3.7 both use postgis-3.3.so)
  2. Extensions to properly find their libraries during upgrades
  3. The multi-version functionality to work correctly

samrose committed:
chore: bump to test

samrose committed:
fix: handling of versions

samrose reviewed (approved):
now passing
